src/components/StoredIdsList.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+interface StoredIdsListProps {
+  ids: string[];
+  currentId?: string;
+  onChange: (selectedId: string) => {};
+}
+
+export function StoredIdsList({
+  ids,
+  currentId,
+  onChange,
+}: StoredIdsListProps) {
+  return (
+    <React.Fragment>
+      <select
+        className="stored-ids-select"
+        onChange={({ target }) => onChange(target.value)}
```

Comment:
maybe onClick is better.. to (re)load the currently selected one.

Reply:
Using `onClick` will introduce a weird behavior that I already tested. Basically, if you click the dropdown and then move over to another part of the page(without actually selecting an option) data will be fetched, which is not very optimal IMHO.

Reply:
Maybe we can keep it on change.. but show the loaded ID.

Reply:
Sure, If I understand correctly that's how it working right now 🙂